✨ [#10] 선택한 경매 상품 상세 정보 조회

puclpu opened the pull request:
## #️⃣연관된 이슈
- 이슈 번호: #10 

## 작업 내용
- 선택한 경매 상품 상세 정보 조회 api 구현
- 선택한 경매 상품 정보를 담는 ProductInfoDto 생성

## 체크리스트
- [x] 코드가 제대로 동작하는지 확인했습니다.
- [x] 문서(코드, 주석, README 등)를 업데이트했습니다.

### 기타
![스크린샷 2024-07-24 오후 4 13 28](https://github.com/user-attachments/assets/bd6d7a2c-f96b-421a-90d1-a58685b05593)

puclpu committed:
✨ [#10] 선택한 경매 상품 상세 정보 조회

- 선택한 경매 상품 상세 정보 조회 api 구현
- 선택한 경매 상품 정보를 담는 ProductInfoDto 생성

je-pa commented on goodsending/src/main/java/com/goodsending/product/service/ProductServiceImpl.java:
```diff
@@ -60,6 +61,34 @@ public ProductCreateResponseDto createProduct(ProductCreateRequestDto requestDto
+
+    Product product = findProduct(productId);
+
+    List<ProductImage> productImageList = findProductImageList(product);
+    List<ProductImageInfoDto> productImages = new ArrayList<>();
+    for (ProductImage productImage : productImageList) {
+      ProductImageInfoDto productImageInfoDto = ProductImageInfoDto.from(productImage);
+      productImages.add(productImageInfoDto);
+    }
```
ProductInfoDto 안에서 구현해주셔도 깔끔해질 것 같아요!

puclpu replied:
해당 코드 위치를 ProductInfoDto로 변경해주었습니다. 감사합니다!

je-pa reviewed (commented):
고생하셨습니다.

puclpu committed:
♻️ [#10] 상품 이미지 형변환 로직 이동

- List<ProductImage>를 List<ProductImageInfoDto>로 변경하는 로직을 Service에서 ProductInfoDto로 이동
